### PR TITLE
Feature/request config timeout

### PR DESCRIPTION
**Pull request by Shooshte**
Proposition on how to handle class method timeouts. Creating apull request to review before adding the decorator to all exposed async class methods

### CODE REVIEW

File: src/layer/AbstractLayer.ts [hunk 1]
```diff
@@ -30,7 +30,12 @@ export class AbstractLayer {
     this.legendUrl = legendUrl;
   }
 
-  public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
+  @timeoutWrapper(2)
+  public async getMap(
+    params: GetMapParams,
+    api: ApiType,
+    requestsConfig?: RequestConfiguration,
```
Comment (sinergise-anze) [inline src/layer/AbstractLayer.ts:37]: It is difficult to review this MR because so many (unrelated) things have changed. Maybe a good idea would be to implement just the timeout functionality, and leave cleanup tasks for another MR, so they can be properly checked too?

File: src/layer/AbstractLayer.ts [hunk 2]
```diff
@@ -30,7 +30,12 @@ export class AbstractLayer {
     this.legendUrl = legendUrl;
   }
 
-  public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
+  @timeoutWrapper(2)
```
Comment (sinergise-anze) [inline src/layer/AbstractLayer.ts:33]: As we have discussed recently, I think it would be better to avoid using the decorators altogether, not make it more explicit how the `reqConfig` parameter changes. 

The suggestion was:
```typescript
public async getMap(params: GetMapParams, api: ApiType, reqConfigWithTimeout?: RequestConfiguration): Promise<Blob> {
  return await ensureTimeout(reqConfigWithTimeout, async (reqConfig) => {
    ...
  })
}
```

Then `ensureTimeout` can look similar to this:
```javascript
async function ensureTimeout(originalReqConfig, innerFunc) {
  if (!originalReqConfig) {
    return await innerFunc(originalReqConfig);
  }

  // setTimeout, clearTimeout,...

  return newReqConfig;
}
```

This would make implementation much easier (imho) and explicit about what is going on. What do you think?

File: README.md [hunk 3]
```diff
@@ -191,100 +204,121 @@ When retrieving an image URL (via `getMapUrl()`) with gain and gamma applied, an
...
+  const tiles = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+} catch (err) {
+  throw err;
+}
+```
 
-To do so a token needs to be created and passed through a requests configuration object. Other config such as retries can also be defined there.
+You can also cancel requests when searching/fetching data. To do so a token needs to be created and passed through the requests configuration object.
 
 In the example below, a cancel token is passed inside the configuration request object. The timeout will cancel the requests after 500 miliseconds, throwing an exception.
```
[the remark this answers is not in the document]
Comment (Shooshte) [reply README.md:263]: Resolving this, opened a [different MR](https://github.com/sentinel-hub/sentinelhub-js/pull/105/files)

**Comment by Shooshte** [comment]
Closing this, opened a new [WIP PR](https://github.com/sentinel-hub/sentinelhub-js/pull/105/files)
